src/kibana/plugins/visualize/editor/editor.js

```diff
@@ -125,6 +125,46 @@ define(function (require) {
 
       editableVis.listeners.click = vis.listeners.click = filterBarClickHandler($state);
       editableVis.listeners.brush = vis.listeners.brush = brushEvent;
+      editableVis.listeners.mapZoomEnd = vis.listeners.mapZoomEnd = function (event) {
```

Comment:
I don't like the idea of sticking map specific code in editor.js. Couldn't this be done in `src/kibana/plugins/vis_types/vislib/tile_map.js`? 

You might take a look at how the `rectangle` event is handled for the tilemap filter here: https://github.com/elastic/kibana/pull/3671/files